components/plugin_marketplace/marketplace_item/marketplace_item.js

```diff
@@ -372,40 +424,55 @@ export default class MarketplaceItem extends React.Component {
             );
         }
 
-        const pluginDetailsInner = (
-            <>
-                {this.props.name} <span className='light subtitle'>{versionLabel}</span>
-                {localTag}
-                <p className={classNames('more-modal__description', {error_text: this.props.error})}>
-                    {this.props.error ? this.props.error : this.props.description}
-                </p>
-            </>
-        );
+        let labelsTags;
+        if (this.props.labels && this.props.labels.length !== 0) {
```

Comment:
nit: is the `length` check redundant here?

Reply:
If `this.props.labels` is `null`, `this.props.labels.length` will throw an error. I'm unsure if the `null` case can happen.

Reply:
is this kinda like a safety check then? what if you do ` labels: PropTypes.array.isRequired,` and then do `(this.props.labels)` here?

Reply:
It's a safety check.

I don't see a reason to require the `labels` field. The current implementation is quite safe, as it catches all cases and doesn't throw an js error if the API is "misused".